daemon/daemon.go

```diff
@@ -951,16 +951,13 @@ func NewDaemon(ctx context.Context, config *config.Config, pluginStore *plugin.S
 		return nil, err
 	}
 
-	// As layerstore initialization may set the driver
-	d.graphDriver = layerStore.DriverName()
-
```

Comment:
With this code removed, what happens if the layer store sets the driver? Is it moved to the image service storage driver?

Reply:
For the old (graph driver) image store, `ImageService.StorageDriver()` is just a proxy for `layerstore.DriverName()`, so the layerstore remains the source of truth; https://github.com/moby/moby/blob/6f8ea5b26e42b4a4f1cd465154d4df0a9f9f880f/daemon/images/service.go#L178-L180

And for snapshotters, it will use the snapshotter name (see https://github.com/moby/moby/pull/43982/files#r948993572 above)